File: parallel-consumer-proxy-clients/parallel-consumer-proxy-client-cpp/src/hello.cpp
// Copyright (C) 2026 Antony Stubbs and contributors

// Prints the one line bin/foreign-client-step.sh checks for. The C++ end of the polyglot build
// scaffolding (astubbs#242); the real client, with its CMake build and its container, replaces it
// when the C++ wave lands - see this module's pom for why neither is here yet.

#include <iostream>

int main() {
    std::cout << "parallel-consumer-proxy-client hello fixture: cpp";
    return 0;
}

File: parallel-consumer-proxy-clients/parallel-consumer-proxy-client-typescript/src/hello.ts
// Copyright (C) 2026 Antony Stubbs and contributors

// Prints the one line bin/foreign-client-step.sh checks for. The TypeScript end of the polyglot
// build scaffolding (astubbs#242) - tsc really compiles this, so the module proves the compiler
// works rather than only that Node can execute a file.

const fixture: string = "parallel-consumer-proxy-client hello fixture: typescript";
process.stdout.write(fixture);
